sdk/storage/azure-storage-blob-v2/src/main/java/com/azure/v2/storage/blob/models/AccessTier.java

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) TypeSpec Code Generator.
+
+package com.azure.v2.storage.blob.models;
+
+import io.clientcore.core.annotations.Metadata;
+import io.clientcore.core.annotations.MetadataProperties;
+import io.clientcore.core.serialization.json.JsonReader;
+import io.clientcore.core.serialization.json.JsonSerializable;
+import io.clientcore.core.serialization.json.JsonToken;
+import io.clientcore.core.serialization.json.JsonWriter;
+import io.clientcore.core.utils.ExpandableEnum;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * The access tiers.
+ */
+public final class AccessTier implements ExpandableEnum<String>, JsonSerializable<AccessTier> {
+    private static final Map<String, AccessTier> VALUES = new ConcurrentHashMap<>();
```

Comment:
Quick thought here, do we want to cache new values? These should be exceptional cases and comparing is as cheap as a `String` comparison, so it should involve a lot of overhead.

If we do choose to not cache, we still will need a cache for well-known constants but this could be simplified to a `HashMap` that is only queried. And possibly in the future have a design without map lookups at all if the set of known constants is small enough where a case statement could be more performant (not likely with string enums but definitely integer ones).

Reply:
One scenario where caching is useful is `==` comparison. Since `==` for normal enums is common, I guess it's good to have this parity for expandable enums.